src/task/CreatePeerConnectionTask.ts

```diff
@@ -187,7 +199,31 @@ export default class CreatePeerConnectionTask extends BaseTask implements Remova
       width = cap.width as number;
       height = cap.height as number;
     }
-    tile.bindVideoStream(attendeeId, false, stream, width, height, streamId);
+    const externalUserId = this.context.realtimeController.realtimeExternalUserIdFromAttendeeId(
+      attendeeId
+    );
+    tile.bindVideoStream(attendeeId, false, stream, width, height, streamId, externalUserId);
+    if (!externalUserId) {
+      const timeoutScheduler = new TimeoutScheduler(10000);
+      const handler = (
+        presentAttendeeId: string,
+        present: boolean,
+        externalUserId: string
+      ): void => {
+        if (attendeeId === presentAttendeeId && present && externalUserId !== null) {
+          tile.bindVideoStream(attendeeId, false, stream, width, height, streamId, externalUserId);
+          this.context.realtimeController.realtimeUnsubscribeToAttendeeIdPresence(handler);
+          this.presenceHandlerSet.delete(handler);
+          timeoutScheduler.stop();
+        }
+      };
+      this.context.realtimeController.realtimeSubscribeToAttendeeIdPresence(handler);
+      this.presenceHandlerSet.add(handler);
+      timeoutScheduler.start(() => {
+        this.removeHandlers();
```

Comment:
Asymmetric lifecycle: before the timer begins we add one handler, once the timer fires, it removes all handlers. Shouldn't it just remove the one handler that was added.